apis/v1/resourceSnapshot_types.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope="Cluster",shortName=rss,categories={fleet-workload}
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:JSONPath=`.metadata.generation`,name="Gen",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ResourceSnapShot is used to store a snapshot of selected resources by a resource placement policy.
+// It is immutable. We may need to produce more than one resourceSnapshot for one ResourcePlacement to
+// get around the 1MB size limit of k8s objects.
+// Each snapshot must have a label "fleet.azure.com/snapshotGroup" with value as the snapshot index.
+// Each must have an annotation "fleet.azure.com/resource-hash" with value as the sha-256 hash
+// value of all the snapshots belong to the same snapshot index.
+type ResourceSnapShot struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of ResourceSnapShot.
+	// +required
+	Spec ResourceSnapShotSpec `json:"spec"`
+
+	// The observed status of ResourceSnapShot.
+	// +optional
+	Status ResourceSnapShotStatus `json:"status,omitempty"`
+}
+
+// ResourceSnapShotSpec	defines the desired state of ResourceSnapShot.
+type ResourceSnapShotSpec struct {
+	// Index is the parent index of this resource snapshot. Each index can have multiple snapshots.
+	// All the snapshots with the same index have the same label "fleet.azure.com/snapshotGroup" that
+	// points to the index.
+	// +required
+	Index int32 `json:"index"`
+
+	// SelectedResources contains a list of resources selected by ResourceSelectors.
+	// +required
+	SelectedResources []ResourceContent `json:"selectedResources"`
+
+	// IsLatest indicates if the resourceSnapshot is the latest or not.
+	// +required
+	IsLatest bool `json:"isLatest"`
```

Comment:
if the resources are split into multiple snapshots, how many snapshots will be set as "latest", only parent or including its children?

Reply:
all the children are marked as "latest".  The "parent" is a logic concept. I just realized that we can use the first snapshot as "parent" and the rest are basically a linked list.

Just thinking out loud. 
My mental model for a "logic" parent is that all the snapshots name starts with the CRP name + index (i.e.  resource-index1-0, resource-index1-1)

for the list model the name will be (resource-index1, resource-index1-1, resource-index1-2)

Given that most of the snapshots will have only 1 object, it seems that the second one is better.